datadog-oci-orm/metrics-setup/function_setup.tf

```
@@ -1,10 +1,36 @@
 resource "null_resource" "Login2OCIR" {
-  provisioner "local-exec" {
-    command = "echo '${var.oci_docker_password}' |  docker login ${local.oci_docker_repository} --username ${local.ocir_namespace}/${var.oci_docker_username} --password-stdin"
-  }
+   provisioner "local-exec" {
+       command = <<EOT
+           #!/bin/bash
+           set -e
+
+           for i in {1..5}; do
+               echo "${var.oci_docker_password}" | docker login ${local.registry_domain} --username ${local.ocir_namespace}/${local.username} --password-stdin && break
+               echo "Retrying Docker login... attempt $i"
+               sleep 10
+           done
+
+           # Check if login was successful
+           if [ $i -eq 5 ]; then
+               echo "Error: Docker login failed after 5 attempts. Trying to login through Oracle Identity Cloud Service"
+               for j in {1..5}; do
+                   echo "${var.oci_docker_password}" | docker login ${local.registry_domain} --username ${local.ocir_namespace}/oracleidentitycloudservice/$username --password-stdin && break
+                   echo "Retrying Docker login through Identity service... attempt $j"
+                   sleep 5
+               done
+               if [ $j -eq 5 ]; then
+                   echo "Error: Docker login failed after 5 attempts."
```

nit ⛏️ : This log line is very similar to the one above it. When read quickly or on a small screen they could appear to be the same
```suggestion
                   echo "Error: Unable to log into Docker. Login failed after an additional 5 attempts."
```